src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -91,16 +115,20 @@
   )
 
   test_target_sources_file_names = sorted(source_root_stripped_test_target_sources.snapshot.files)
+  timeout_seconds = calculate_timeout_seconds(
+    timeout_disabled=pytest.options.timeout_disabled,
+    target_timeout=getattr(test_target, 'timeout', None),
+    timeout_default=pytest.options.timeout_default,
+    timeout_maximum=pytest.options.timeout_maximum,
```

Comment:
Should the `pytest` subsystem have a mixin that adds these timeout values (similar to TestRunnerTaskMixin, but for Subsystems instead)? You could then define the `calculate_timeout_seconds` method as part of that mixin.

Reply:
> Should the pytest subsystem have a mixin

Likely, but I think this may be premature to generalize. Once we add another test runner, we can figure out the best way to generalize out timeout support.

Reply:
Fine with that since this is green. But over time we should be leaning more toward assuming other test runners (ie, it's not premature generalization: we have known consumers and a parallel universe here).

Reply:
> But over time we should be leaning more toward assuming other test runners (ie, it's not premature generalization: we have known consumers and a parallel universe here).

Agreed that it's not premature in terms of the _need_ for it, my point is more in the best way to _implement_ it, such as my example above about where to put the thing.

Another thing I've been considering is if mixins are still the best mechanism for DRYing option registration? An alternative could be a top-level function that takes a `register` and mutates it to register those new options, for example. That would allow us to do things like pass args to the function so that we could do something like make the `--help` message specific to each subsystem, rather than generic. (We use specific `--help` messages for the `--args` options currently to give examples of what `--args` you might pass to Isort vs. Black vs. Flake8 vs. Pytest, and I think it's very helpful).